[18.09] makemkv: 1.12.3 -> 1.14.1

nyanloutre opened the pull request:
###### Motivation for this change

Previous URL is unreachable

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

danieldk committed:
makemkv: 1.12.3 -> 1.14.0

Changes: https://www.makemkv.com/download/history.html

Since this derivation only support Linux on x86_64, set the platform
meta data accordingly.

(cherry picked from commit f528731fe9724056c706e30e17ecb299707e17e9)

nyanloutre committed:
makemkv: 1.14.0 -> 1.14.1

(cherry picked from commit 30bc36c64e11ca8d7ad3488a3708fce6a6ef57ea)

c0bw3b reviewed (changes requested):
See #50437 for a stable fix to the moving URLs issue

nyanloutre committed:
makemkv: using two URLs for src

This will prevent future failures due to the first URL being broken each
time a new version is released

See https://github.com/NixOS/nixpkgs/pull/50437#issuecomment-439329862
for details

(cherry picked from commit 05166dcce12118b167910f5149d1407826c9ecd5)

nyanloutre commented:
@c0bw3b the fix is backported too

c0bw3b commented:
Thanks! Now we have a stable package in the release
